### DISCUSSION TITLE
Is pyMuPDF compatable with PDF 1.6?

### DISCUSSION THREAD

**Discussion by babazaroni** [pymupdf/PyMuPDF#3923] [Looking for help]
I have two similar pdfs.  One has format PDF 1.4, while the other has 1.6.   They have the same content, but get_text() works only with 1.4.   Get_text() on the pdf 1.6 returns an empty string?  Any ideas?

**Comment by JorjMcKie** [comment]
Let's start with actually looking at them ... 😎
PyMuPDF supports all PDF versions.

**Reply by babazaroni** [reply]
Unfortunately, they are sensitive financial reports.  I wonder if the institution decided to build the pdfs so get_text() cannot be used on them.

**Reply by JorjMcKie** [reply]
Well then all I can say is that they definitely do not have the "same content".
The are possibilities to prevent text extraction, but they all can be circumvented too (also by using PyMuPDF).
Pages may be images for example, or the text may not be embedded but encoded as little vector graphics, like instead of writing the capital letter "A" three lines are drawn "/", "-", "\", letter "o" is drawn as an ellipse, etc.
Here are two examples made with PyMuPDF ... "same content", but only the first one is extractable.
[loremipsum.pdf](https://github.com/user-attachments/files/17288965/loremipsum.pdf)
[loremipsum-svg.pdf](https://github.com/user-attachments/files/17288968/loremipsum-svg.pdf)
